test/ResultSet/ResultSetIntegrationTest.php

```diff
@@ -90,6 +90,20 @@ public function testCanProvideIteratorAsDataSource()
         $this->assertSame($it, $this->resultSet->getDataSource());
     }
 
+    public function testCanProvideArrayAsDataSource()
+    {
+        $dataSource = [['foo']];
+        $this->resultSet->initialize($dataSource);
+        $this->assertEquals($dataSource[0], (array) $this->resultSet->current());
+
+        $returnType = new ReturnType();
+        $dataSource = [$returnType];
+        $this->resultSet->setArrayObjectPrototype($returnType);
```

Comment:
Thanks for your feedback. I don't think your suggestion covers the use case, while using ArrayObject obfuscates what I'm trying to prove, as ArrayObject is used all over ResultSet and it seems to know very well how to handle it. Problems arise when you are not using an ArrayObject, so I would rather not venture into using it. Also, when I try it, the test case `testCanProvideArrayAsDataSource` fails with `Object must be of type ArrayObject, or at least implement exchangeArray`.